src/fluree/db/util/context.cljc

```diff
@@ -0,0 +1,105 @@
+(ns fluree.db.util.context
+  (:require [clojure.string :as str]))
+
+;; handles some default context merging.
+
+#?(:clj (set! *warn-on-reflection* true))
+
+(defn mapify-context
+  "An unparsed context may be a vector of maps or URLs. This merges them together into one large map.
+
+  If a sequence context is used, and an item is an empty string, will substitute a default context."
+  [context default-context]
+  (when context
+    (cond
+
+      (map? context)
+      context
+
+      (sequential? context)
+      (reduce (fn [acc context-item]
+                (cond
+
+                  (map? context-item)
+                  (merge acc context-item)
+
+                  (= "" context-item)
+                  (if default-context
+                    (merge acc default-context)
+                    (throw (ex-info (str "Context uses a default context with empty string (''), "
+                                         "but not default context provided.")
+                                    {:status 400
+                                     :error  :db/invalid-context})))
+
+                  :else
+                  (throw (ex-info (str "Only context maps are supported at the moment, provided: " context-item)
+                                  {:status 400
+                                   :error  :db/invalid-context}))))
+              {} context)
+
+      :else
+      (throw (ex-info (str "Invalid context provided: " context)
+                      {:status 400
+                       :error  :db/invalid-context})))))
+
+
+(defn stringify-context
+  "Ensures mapified context that might use keyword keys is in string format."
+  [context]
+  (when context
+    (if (map? context)
+      (reduce-kv
+        (fn [acc k v]
```

Comment:
Is it ever possible to have keywords in the values of a context? Or just top-level keys?

I'm thinking of contexts with nesting like:
```


{
  "@context": {
    "name": "http://schema.org/name",
    "image": {
      "@id": "http://schema.org/image",
      "@type": "@id"
    },
    "homepage": {
      "@id": "http://schema.org/url",
      "@type": "@id"
    }
  }
}

Reply:
You are absolutely correct, this code is naive. I'll update it to be a little less naive in the PR to update default context I'm about to make (fluree/core issue 8).

I'm not sure how robust it needs to be, so I'll try to correct the issue you point out without going too far down the rabbit hole - given that keyword contexts are not standard and a convenience for CLJ devs, who will always have the ability to use the true standards-based context if needed.

Reply:
Sounds good. 

I do think at some point we should have a "warning" of sorts that keyword contexts are experimental/a convenience (in our docs or readme). I think those expectations should be set appropriately.